File: 609-find-duplicate-file-in-system/609-find-duplicate-file-in-system.js
/**
 * @param {string[]} paths
 * @return {string[][]}
 */
var findDuplicate = function(paths) {
    var mapper=[], res=[], k=0;
    paths.map((e,i)=>{
        var tokens = e.split(' ');
        for(var i=1; i<tokens.length; i++){
            var f = tokens[i].split('(');   // [1.txt, abcd)]
             mapper[f[1]]=mapper[f[1]]??k++;
            if(!res[mapper[f[1]]])
                res[mapper[f[1]]]=[];
            res[mapper[f[1]]].push(tokens[0]+'/'+f[0]);
            // console.log(res[mapper[f[1]]]);
        }
    })
    // console.log(res);
    return res.filter(e=>{
        return e.length>=2;
    });
};

File: 609-find-duplicate-file-in-system/609-find-duplicate-file-in-system.cpp
class Solution {
public:
    vector<vector<string>> findDuplicate(vector<string>& paths) {
        vector<vector<string>> v;
        map <string, int> m;
        int k=1;
        for(auto p: paths){
            stringstream ss(p);
            string t;
            vector <string> dir = {};
            while(!ss.eof()){
                getline(ss, t, ' ');
                dir.push_back(t);
            }
            for(int i=1; i<dir.size(); i++){
                stringstream ss(dir[i]);
                string x;
                vector <string> tok = {};
                while(!ss.eof()){
                    getline(ss, x, '(');
                    tok.push_back(x);
                }
                if(!m[tok[1]])
                    m[tok[1]]=k++;
                if(m[tok[1]]>v.size())
                    v.push_back({});
                v[m[tok[1]]-1].push_back(dir[0]+'/'+tok[0]);
            }
        }
        vector<vector<string>>res;
        for(auto vec: v)
            if(vec.size()>1)
                res.push_back(vec);
        return res;
    }
};